src/displayapp/screens/settings/SettingWakeUp.cpp

```diff
@@ -85,27 +85,46 @@ bool SettingWakeUp::Refresh() {
 }
 
 void SettingWakeUp::UpdateSelected(lv_obj_t* object, lv_event_t event) {
-  if (event == LV_EVENT_VALUE_CHANGED) {
+  if (event == LV_EVENT_VALUE_CHANGED && !isLocked) {
+    isLocked = true;
```

Comment:
Why do you need this variable `isLocked`?

`SettingWakeUp::UpdateSelected()` will never be called from 2 different tasks (this callback is called by LVGL which only runs in the context of DisplayApp) so it should not be needed to protect against race conditions here.

Reply:
later on we use `lv_checkebox_set_checked()`, so isLocked is there to prevent those extra events from causing issues

Reply:
I still doesn't look good to me... I mean... couldn't we find another way to implement the logic so that we don't have to protect ourselves again event we generate ?

Reply:
Another idea I've just had : would it be possible to explicitly disable the event on the checkboxes while we are changing their states, and re-enabling it once we are done?

Reply:
My concern about this is that the code is confusing. When I read the code the first time, when I saw the variable `IsLocked`, I thought this was a simple implementation of a mutex to protect against race conditions against multiple tasks. And then, as I said before, it makes little sense to protect ourselves against event we generate ourselves.

I'm just trying to see if there is any better solution that would make the code more expressive and readable..

If we cannot find a better solution right now, I think we should at least find a better name for IsLocked and document why it's needed so that someone can try to improve that in the future.

Reply:
If I understood correctly, `SettingWakeUp::UpdateSelected()` is called whenever the state of a checkbox changes. This change happens when the user taps a checkbox or... when the code explicitly changes the state of the checkbox.
That's what happens here : when the user taps a checkbox, the code needs to uncheck  mutually exclusives checkboxes. This call to `lv_checkbox_set_state()` will trig another (not needed) call to `SettingWakeUp::UpdateSelected()`.

If we decide to keep this workflow, I would just like to rename `isLocked` with something that shows what it's actually doing : ignore event. Maybe `ignoreSelectEvent` ?

Reply:
I don't think that's correct. I even looked at the lvgl sources and couldn't find any evidence of an event being generated when `lv_checkbox_set_state` is called. It wouldn't really make sense either since there is `lv_event_send(obj, LV_EVENT_..., &custom_data)` if we needed to simulate an actual tap.

I would just try this, but I'm worried that messing with settings could cause problems on my sealed pinetime.